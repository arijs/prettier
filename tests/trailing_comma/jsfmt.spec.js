run_spec(__dirname, null, ["typescript"]);
run_spec(__dirname, { trailingComma: "all" }, ["typescript"]);
run_spec(__dirname, { trailingComma: "es5" }, ["typescript"]);
run_spec(__dirname, { trailingComma: "array,import" }, ["typescript"]);
run_spec(__dirname, { trailingComma: "object,export,arguments" }, ["typescript"]);
