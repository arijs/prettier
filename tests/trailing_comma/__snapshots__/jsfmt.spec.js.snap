// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`es5.js 1`] = `
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to);
  return "nothing";
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(
    app,
    email_id,
    email_address,
    subject,
    html,
    reply_to
  );
  return "nothing";
}

`;

exports[`es5.js 2`] = `
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to);
  return "nothing";
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to,
) {
  send_single_email_implementation(
    app,
    email_id,
    email_address,
    subject,
    html,
    reply_to,
  );
  return "nothing";
}

`;

exports[`es5.js 3`] = `
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to);
  return "nothing";
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(
    app,
    email_id,
    email_address,
    subject,
    html,
    reply_to
  );
  return "nothing";
}

`;

exports[`es5.js 4`] = `
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to);
  return "nothing";
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(
    app,
    email_id,
    email_address,
    subject,
    html,
    reply_to
  );
  return "nothing";
}

`;

exports[`es5.js 5`] = `
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to
) {
  send_single_email_implementation(  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to);
  return "nothing";
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function send_single_email(
  app,
  email_id,
  email_address,
  subject,
  html,
  reply_to,
) {
  send_single_email_implementation(
    app,
    email_id,
    email_address,
    subject,
    html,
    reply_to,
  );
  return "nothing";
}

`;

exports[`export.js 1`] = `
export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
export {
  value1,
  value2 as value2_renamed,
  value3,
  value4 as value4_renamed,
  value5
} from "exports";

`;

exports[`export.js 2`] = `
export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
export {
  value1,
  value2 as value2_renamed,
  value3,
  value4 as value4_renamed,
  value5,
} from "exports";

`;

exports[`export.js 3`] = `
export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
export {
  value1,
  value2 as value2_renamed,
  value3,
  value4 as value4_renamed,
  value5,
} from "exports";

`;

exports[`export.js 4`] = `
export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
export {
  value1,
  value2 as value2_renamed,
  value3,
  value4 as value4_renamed,
  value5
} from "exports";

`;

exports[`export.js 5`] = `
export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
export {
  value1,
  value2 as value2_renamed,
  value3,
  value4 as value4_renamed,
  value5,
} from "exports";

`;

exports[`function-calls.js 1`] = `
const a = (param1, param2, param3) => {}

a('value', 'value2', 'value3');

a(
  'a-long-value',
  'a-really-really-long-value',
  'a-really-really-really-long-value',
);

a('value', 'value2', a('long-nested-value', 'long-nested-value2', 'long-nested-value3'));
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = (param1, param2, param3) => {};

a("value", "value2", "value3");

a(
  "a-long-value",
  "a-really-really-long-value",
  "a-really-really-really-long-value"
);

a(
  "value",
  "value2",
  a("long-nested-value", "long-nested-value2", "long-nested-value3")
);

`;

exports[`function-calls.js 2`] = `
const a = (param1, param2, param3) => {}

a('value', 'value2', 'value3');

a(
  'a-long-value',
  'a-really-really-long-value',
  'a-really-really-really-long-value',
);

a('value', 'value2', a('long-nested-value', 'long-nested-value2', 'long-nested-value3'));
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = (param1, param2, param3) => {};

a("value", "value2", "value3");

a(
  "a-long-value",
  "a-really-really-long-value",
  "a-really-really-really-long-value",
);

a(
  "value",
  "value2",
  a("long-nested-value", "long-nested-value2", "long-nested-value3"),
);

`;

exports[`function-calls.js 3`] = `
const a = (param1, param2, param3) => {}

a('value', 'value2', 'value3');

a(
  'a-long-value',
  'a-really-really-long-value',
  'a-really-really-really-long-value',
);

a('value', 'value2', a('long-nested-value', 'long-nested-value2', 'long-nested-value3'));
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = (param1, param2, param3) => {};

a("value", "value2", "value3");

a(
  "a-long-value",
  "a-really-really-long-value",
  "a-really-really-really-long-value"
);

a(
  "value",
  "value2",
  a("long-nested-value", "long-nested-value2", "long-nested-value3")
);

`;

exports[`function-calls.js 4`] = `
const a = (param1, param2, param3) => {}

a('value', 'value2', 'value3');

a(
  'a-long-value',
  'a-really-really-long-value',
  'a-really-really-really-long-value',
);

a('value', 'value2', a('long-nested-value', 'long-nested-value2', 'long-nested-value3'));
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = (param1, param2, param3) => {};

a("value", "value2", "value3");

a(
  "a-long-value",
  "a-really-really-long-value",
  "a-really-really-really-long-value"
);

a(
  "value",
  "value2",
  a("long-nested-value", "long-nested-value2", "long-nested-value3")
);

`;

exports[`function-calls.js 5`] = `
const a = (param1, param2, param3) => {}

a('value', 'value2', 'value3');

a(
  'a-long-value',
  'a-really-really-long-value',
  'a-really-really-really-long-value',
);

a('value', 'value2', a('long-nested-value', 'long-nested-value2', 'long-nested-value3'));
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = (param1, param2, param3) => {};

a("value", "value2", "value3");

a(
  "a-long-value",
  "a-really-really-long-value",
  "a-really-really-really-long-value",
);

a(
  "value",
  "value2",
  a("long-nested-value", "long-nested-value2", "long-nested-value3"),
);

`;

exports[`import.js 1`] = `
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from '.';
import {fitsIn, oneLine} from '.';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from ".";
import { fitsIn, oneLine } from ".";

`;

exports[`import.js 2`] = `
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from '.';
import {fitsIn, oneLine} from '.';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns,
} from ".";
import { fitsIn, oneLine } from ".";

`;

exports[`import.js 3`] = `
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from '.';
import {fitsIn, oneLine} from '.';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns,
} from ".";
import { fitsIn, oneLine } from ".";

`;

exports[`import.js 4`] = `
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from '.';
import {fitsIn, oneLine} from '.';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns,
} from ".";
import { fitsIn, oneLine } from ".";

`;

exports[`import.js 5`] = `
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from '.';
import {fitsIn, oneLine} from '.';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import {
  runTaskForChanged,
  description,
  someOtherLabel,
  thatMakes,
  itGo,
  multiLine,
  andMore,
  soWeCanGetItTo80Columns
} from ".";
import { fitsIn, oneLine } from ".";

`;

exports[`object.js 1`] = `
const a = {
  b: true,
  c: {
    c1: 'hello'
  },
  d: false
};

const aLong = {
  bHasALongName: 'a-long-value',
  cHasALongName: {
    c1: 'a-really-long-value',
    c2: 'a-really-really-long-value',
  },
  dHasALongName: 'a-long-value-too'
};

const aLong = {
  dHasALongName: 'a-long-value-too',
  eHasABooleanExpression: a === a,
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = {
  b: true,
  c: {
    c1: "hello"
  },
  d: false
};

const aLong = {
  bHasALongName: "a-long-value",
  cHasALongName: {
    c1: "a-really-long-value",
    c2: "a-really-really-long-value"
  },
  dHasALongName: "a-long-value-too"
};

const aLong = {
  dHasALongName: "a-long-value-too",
  eHasABooleanExpression: a === a
};

`;

exports[`object.js 2`] = `
const a = {
  b: true,
  c: {
    c1: 'hello'
  },
  d: false
};

const aLong = {
  bHasALongName: 'a-long-value',
  cHasALongName: {
    c1: 'a-really-long-value',
    c2: 'a-really-really-long-value',
  },
  dHasALongName: 'a-long-value-too'
};

const aLong = {
  dHasALongName: 'a-long-value-too',
  eHasABooleanExpression: a === a,
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = {
  b: true,
  c: {
    c1: "hello",
  },
  d: false,
};

const aLong = {
  bHasALongName: "a-long-value",
  cHasALongName: {
    c1: "a-really-long-value",
    c2: "a-really-really-long-value",
  },
  dHasALongName: "a-long-value-too",
};

const aLong = {
  dHasALongName: "a-long-value-too",
  eHasABooleanExpression: a === a,
};

`;

exports[`object.js 3`] = `
const a = {
  b: true,
  c: {
    c1: 'hello'
  },
  d: false
};

const aLong = {
  bHasALongName: 'a-long-value',
  cHasALongName: {
    c1: 'a-really-long-value',
    c2: 'a-really-really-long-value',
  },
  dHasALongName: 'a-long-value-too'
};

const aLong = {
  dHasALongName: 'a-long-value-too',
  eHasABooleanExpression: a === a,
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = {
  b: true,
  c: {
    c1: "hello",
  },
  d: false,
};

const aLong = {
  bHasALongName: "a-long-value",
  cHasALongName: {
    c1: "a-really-long-value",
    c2: "a-really-really-long-value",
  },
  dHasALongName: "a-long-value-too",
};

const aLong = {
  dHasALongName: "a-long-value-too",
  eHasABooleanExpression: a === a,
};

`;

exports[`object.js 4`] = `
const a = {
  b: true,
  c: {
    c1: 'hello'
  },
  d: false
};

const aLong = {
  bHasALongName: 'a-long-value',
  cHasALongName: {
    c1: 'a-really-long-value',
    c2: 'a-really-really-long-value',
  },
  dHasALongName: 'a-long-value-too'
};

const aLong = {
  dHasALongName: 'a-long-value-too',
  eHasABooleanExpression: a === a,
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = {
  b: true,
  c: {
    c1: "hello"
  },
  d: false
};

const aLong = {
  bHasALongName: "a-long-value",
  cHasALongName: {
    c1: "a-really-long-value",
    c2: "a-really-really-long-value"
  },
  dHasALongName: "a-long-value-too"
};

const aLong = {
  dHasALongName: "a-long-value-too",
  eHasABooleanExpression: a === a
};

`;

exports[`object.js 5`] = `
const a = {
  b: true,
  c: {
    c1: 'hello'
  },
  d: false
};

const aLong = {
  bHasALongName: 'a-long-value',
  cHasALongName: {
    c1: 'a-really-long-value',
    c2: 'a-really-really-long-value',
  },
  dHasALongName: 'a-long-value-too'
};

const aLong = {
  dHasALongName: 'a-long-value-too',
  eHasABooleanExpression: a === a,
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = {
  b: true,
  c: {
    c1: "hello",
  },
  d: false,
};

const aLong = {
  bHasALongName: "a-long-value",
  cHasALongName: {
    c1: "a-really-long-value",
    c2: "a-really-really-long-value",
  },
  dHasALongName: "a-long-value-too",
};

const aLong = {
  dHasALongName: "a-long-value-too",
  eHasABooleanExpression: a === a,
};

`;

exports[`trailing_whitespace.js 1`] = `
let example = [
  'FOO',
  'BAR',
  // Comment
];

foo({},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a
}

this.getAttribute(function(s)
  /*string*/ {
  console.log()
});
this.getAttribute(function(s) /*string*/ {
  console.log()
});
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let example = [
  "FOO",
  "BAR"
  // Comment
];

foo(
  {}
  // Comment
);

o = {
  state
  // Comment
};

o = {
  state

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}

this.getAttribute(function(s) /*string*/ {
  console.log();
});
this.getAttribute(function(s) /*string*/ {
  console.log();
});

`;

exports[`trailing_whitespace.js 2`] = `
let example = [
  'FOO',
  'BAR',
  // Comment
];

foo({},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a
}

this.getAttribute(function(s)
  /*string*/ {
  console.log()
});
this.getAttribute(function(s) /*string*/ {
  console.log()
});
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let example = [
  "FOO",
  "BAR",
  // Comment
];

foo(
  {},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a;
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a;
}

this.getAttribute(function(s) /*string*/ {
  console.log();
});
this.getAttribute(function(s) /*string*/ {
  console.log();
});

`;

exports[`trailing_whitespace.js 3`] = `
let example = [
  'FOO',
  'BAR',
  // Comment
];

foo({},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a
}

this.getAttribute(function(s)
  /*string*/ {
  console.log()
});
this.getAttribute(function(s) /*string*/ {
  console.log()
});
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let example = [
  "FOO",
  "BAR",
  // Comment
];

foo(
  {}
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}

this.getAttribute(function(s) /*string*/ {
  console.log();
});
this.getAttribute(function(s) /*string*/ {
  console.log();
});

`;

exports[`trailing_whitespace.js 4`] = `
let example = [
  'FOO',
  'BAR',
  // Comment
];

foo({},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a
}

this.getAttribute(function(s)
  /*string*/ {
  console.log()
});
this.getAttribute(function(s) /*string*/ {
  console.log()
});
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let example = [
  "FOO",
  "BAR",
  // Comment
];

foo(
  {}
  // Comment
);

o = {
  state
  // Comment
};

o = {
  state

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a;
}

this.getAttribute(function(s) /*string*/ {
  console.log();
});
this.getAttribute(function(s) /*string*/ {
  console.log();
});

`;

exports[`trailing_whitespace.js 5`] = `
let example = [
  'FOO',
  'BAR',
  // Comment
];

foo({},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB
  // Comment
) {
  a
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a
}

this.getAttribute(function(s)
  /*string*/ {
  console.log()
});
this.getAttribute(function(s) /*string*/ {
  console.log()
});
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let example = [
  "FOO",
  "BAR"
  // Comment
];

foo(
  {},
  // Comment
);

o = {
  state,
  // Comment
};

o = {
  state,

  // Comment
};

function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a;
}
function supersupersupersuperLongF(
  supersupersupersuperLongA,
  supersupersupersuperLongB,
  // Comment
) {
  a;
}

this.getAttribute(function(s) /*string*/ {
  console.log();
});
this.getAttribute(function(s) /*string*/ {
  console.log();
});

`;
